src/core/lib/security/credentials/external/external_account_credentials.cc

```diff
@@ -347,10 +348,11 @@ void ExternalAccountCredentials::ExchangeToken(
         absl::StrFormat("%s:%s", options_.client_id, options_.client_secret);
     char* encoded_cred =
         grpc_base64_encode(raw_cred.c_str(), raw_cred.length(), 0, 0);
```

Comment:
Suggest changing this to use `absl::Base64Escape()`, so that you don't need to use C-style strings that need to be manually freed.